src/client/helpers.js

```diff
@@ -12,6 +12,45 @@ const failCodes = {
   500: 'Internal Server Error',
   503: 'Service Unavailable',
 };
+
+/**
+ * Gets a response header, switching between development/test modalities and production
+ *
+ * @param {Object} response - Response object.
+ * @param {string} target - The target to retrieve.
+ * @returns {Object} - Returns object in response header.
+ */
+function responseHeaderRetrieve(response, target) {
+  if(['development', 'test'].includes(process.env.NODE_ENV)) {
+    return response.headers[target];
+  }
+  return response.headers.get(target);
+}
+function responseGetStatus(response) {
+  if(['development', 'test'].includes(process.env.NODE_ENV)) {
+    return response.statusCode;
+  }
+  return response.status;
+}
+function responseStatusText(response) {
+  if(['development', 'test'].includes(process.env.NODE_ENV)) {
+    return response.statusMessage;
+  }
+  return response.statusText;
+
+}
+/**
+ * Gets a response body json, switching between development/test modalities and production
+ *
+ * @param {Object} response - Response object.
+ * @returns {Object} - Target header object.
+ */
+function responseBodyJSON(response) {
+  if(['development', 'test'].includes(process.env.NODE_ENV)) {
+    return JSON.parse(response.body);
+  }
+  return response.json();
+}
```

Comment:
🤔  This breaks production development parity. Why is this required? How about parsing the body in all env? (line 50)

Reply:
Nock  fixtures only work with request lib.  Sell all the machinations necessary to employ request instead of native request in node.js

Reply:
Is there a transparent record mode / playback mode in msw?
IE logically:
record(fixturname) {
if fixture found use existing fixture
else
     allow request to go through
do work/test
}

Reply:
I might be wrong and I admit I don't know the library, but I find it strange that a production ready library requires us to manually write all this untestable code.

Reply:
> Is there a transparent record mode / playback mode in msw?

yes, it's like the VCR gem. It's been 4 years that I don't touch it, but I remember being very happy using it.

Reply:
> I might be wrong and I admit I don't know the library, but I find it strange that a production ready library requires us to manually write all this untestable code.

Please help by integrating msw, and rip out the conditional code, keeping the above `notional` form factor of
record() {
if fixture found use fixture
else allow request to flow through and capture

do work/test
end record
}

Reply:
Node version I use is 18.4

Reply:
💣  - My memory failed me - MSW doesn't allow you to record missing calls automatically. You need to do that manually 😞